tests/local/power_bi/test_power_bi.py

```diff
@@ -23,103 +23,1089 @@ def test_get_workspace_success(self, mock_get):
+                    "refreshAttempts": None,
+                },
+                {
+                    "id": 1,
+                    "refreshType": "OnDemand",
+                    "status": "Completed",
+                    "startTime": "2024-02-26T10:00:00Z",
+                    "endTime": "2024-02-26T10:05:00Z",  # winter time: 1 hour
+                    "serviceExceptionJson": None,  # difference from UTC
+                    "requestId": "74d25c0b-0473-4dd9-96ff-3ca737b072a7",
+                    "refreshAttempts": None,
+                },
+            ]
+        }
+        get_reimbursement_refresh_history = {
+            "value": [
+                {
+                    "id": 2,
+                    "refreshType": "ViaEnhancedApi",
+                    "status": "Completed",
+                    "startTime": "2024-02-24T10:00:00Z",
+                    "endTime": "2024-02-24T10:09:00Z",
+                    "serviceExceptionJson": None,
+                    "requestId": "8d315d90-6105-450b-af2b-8654b962db98",
+                    "refreshAttempts": None,
+                },
+                {
+                    "id": 1,
+                    "refreshType": "ViaApi",
+                    "status": "Completed",
+                    "startTime": "2024-02-23T10:00:00Z",
+                    "endTime": "2024-02-23T10:10:00Z",
+                    "serviceExceptionJson": None,
+                    "requestId": "11abf327-6e6b-4cdf-a9a0-f443d83680ec",
+                    "refreshAttempts": None,
+                },
+            ]
+        }
+
+        def requests_get(url, headers):
+            mock_response = Mock()
+            mock_response.status_code = 200
+            mock_response.json.return_value = None
+            if headers == "api_header":
+                if url.endswith("groups"):
+                    mock_response.json.return_value = get_workspaces
+                elif (
+                    url.endswith("datasets")
+                    and "614850c2-3a5c-4d2d-bcaa-d3f20f32a2e0" in url
+                ):
+                    mock_response.json.return_value = get_finance_datasets
+                elif (
+                    url.endswith("refreshes")
+                    and "b1f0a07e-e348-402c-a2b2-11f3e31181ce" in url
+                ):
+                    mock_response.json.return_value = get_invoicing_refresh_history
+                elif (
+                    url.endswith("refreshes")
+                    and "2e848e9a-47a3-4b0e-a22a-af35507ec8c4" in url
+                ):
+                    mock_response.json.return_value = get_reimbursement_refresh_history
+                elif (
+                    url.endswith("refreshes")
+                    and "4869cc38-646c-45b6-89fd-62615beb9853" in url
+                ):
+                    mock_response.status_code = 401
+                if mock_response.json.return_value or mock_response.status_code != 200:
+                    return mock_response
```

Comment:
Same comment as earlier on the change that nothing is returned?

Reply:
An exception is cast as earlier:

    raise ValueError("Unknown URL! " + url)